src/wallet3/keyring.hpp

```diff
@@ -123,6 +129,8 @@ namespace wallet
 
 
     hw::core::device_default key_device;
+    std::unordered_map<crypto::public_key, cryptonote::subaddress_index> subaddresses;
+    std::vector<std::vector<std::string>> subaddress_labels;
```

Comment:
This (the labels, I mean) feels like less of a Keyring detail and more of a Wallet detail, and something that will need to be in the database.